core/scripts/cli/sponsor/viewLiquidationDetails.js

```diff
@@ -1,24 +1,120 @@
+    choices
+  });
+  if (input["choice"] === backChoice) {
+    return;
+  }
+
+  await viewWithdrawRewardsMenu(web3, artifacts, emp, liquidationStructs[input["choice"]], input["choice"]);
+};
+
+/**
+ * @notice Display menu of options to withdraw rewards from eligible liquidations.
+ */
+const viewWithdrawRewardsMenu = async (web3, artifacts, emp, liquidation, id) => {
+  // If liquidation `sponsor` property is empty, then either the sponsor's liquidation rewards have been withdrawn or the
+  // liquidation data has been deleted following an expired liquidation or a failed dispute.
+  // Therefore, this should catch all liquidations with state == `UNINITIALIZED` following a deletion of the liquidation struct.
+  if (liquidation.sponsor === "0x0000000000000000000000000000000000000000") {
+    console.log("There are no rewards to withdraw from this liquidation");
+    return;
+  }
 
+  // Now we know that the liquidation still has rewards to be withdrawn, but we need to check if the sponsor is eligible
+  // to receive rewards.
+  const ExpandedERC20 = artifacts.require("ExpandedERC20");
   const sponsorAddress = await getDefaultAccount(web3);
-  const display = `Liquidated at epoch time ${liquidation.liquidationTime} by ${liquidation.liquidator}`;
+  const liquidationTimeReadable = new Date(Number(liquidation.liquidationTime.toString()) * 1000);
+  const display = `Liquidated at time ${liquidationTimeReadable} by ${liquidation.liquidator}`;
   const backChoice = "Back";
-  const withdrawAction = "Withdraw";
+  // Sponsors can withdraw rewards only if a liquidation has been disputed successfully.
+  const withdrawAction = "Dispute succeeded; withdraw rewards";
   choices = [{ name: backChoice }];
+
   // Check if the sponsor can withdraw by seeing if `withdrawLiquidation` reverts.
   try {
     await emp.withdrawLiquidation.call(id, sponsorAddress);
     choices.push({ name: withdrawAction });
   } catch (err) {
-    // Withdraw wouldn't work so it shouldn't be a valid option.
+    // Withdraw wouldn't work so it shouldn't be a valid option. Let's print out a detailed response to the user to indicate why
+    // `withdraw` will fail.
+
+    // The sponsor can only withdraw rewards from a successfully disputed liquidation.
+    if (liquidation.state === LiquidationStatesEnum.PRE_DISPUTE) {
+      // If liquidation state is PRE_DISPUTE and `withdrawLiquidation` fails, then the liquidation has either expired
+      // or is pre-expiry. Only a liquidator can withdraw from an expired liquidation.
+      console.log(
+        "Cannot withdraw rewards from a liquidation that is pre-expiry and has not been disputed, or has already expired"
+      );
+    } else if (liquidation.state === LiquidationStatesEnum.PENDING_DISPUTE) {
+      // If the liquidation state is PENDING_DISPUTE and `withdrawLiquidation` fails, then it indicates that either
+      // a price has not resolved yet, or a price has resolved that indicates that the dispute will FAIL.
```

Comment:
This is a weird (but totally valid) case. Is there any way to tell the difference between the case where the oracle has a price and the case where it doesn't?

Reply:
Done, PTAL at the logic. I fetch Oracle from finder and check whether it has a price. If it does then that means the dispute will fail and we can change the messaging.